File: PASCAL/Cap06/c06ex04.pas
program c06ex04;

var
  A: array of string;
  N, I: Integer;

begin
  Write('Entre a quantidade de elementos da matriz: ');
  ReadLn(N);
  WriteLn;

  SetLength(A, N);

  for I := 1 to N do
    begin
      Write('Entre o ', I:3, 'o. nome: ');
      ReadLn(A[I]);
    end;

  WriteLn;
  WriteLn('Foram fornecidos os nomes:');
  WriteLn;

  for I := 1 to N do
    begin
      WriteLn('Nome ', I:3, ' = ', A[I]);
    end;

  WriteLn;
  Write('Tecle <Enter> para encerrar... ');
  ReadLn;
end.

File: KOTLIN/Cap06/c06ex04.kt
// c06ex04.kt

fun main() {
    var N: Int
    val A = mutableListOf<String>()

    print("Entre a quantidade de elementos da matriz: ")
    N = readln().toInt()
    println()

    for (i in 0..(N - 1)) {
        print("Entre o ${i + 1}o. nome: ")
        A.add(readln())
    }

    println()
    println("Foram fornecidos os nomes:")
    println()

    for (i in 0..(N - 1))
        println("Nome ${i + 1} = ${A[i]}")

    print("\nTecle <Enter> para encerrar... ")
    readln()
}
